### DISCUSSION TITLE
`Portal` of `@radix-ui/react-alert-dialog` has so many attributes not described in docs.

### DISCUSSION THREAD

**Discussion by sudoaugustin** [radix-ui/primitives#1956] [Help]
My typescript shows a list of options not stated in the document. However, the `className` of my `Portal` don't get rendered as `Portal` seems to be `Fragment`. I don't know whether it's a bug or documentation needs.

<img width="645" alt="Screenshot 2023-02-15 at 12 31 20 AM" src="https://user-images.githubusercontent.com/57541158/218819860-120f9c6d-5877-4dfb-8a14-717120842179.png">

**Comment by benoitgrelard** [comment]
Hey @sudoaugustin, Thanks for bringing this to our attention!
You are correct, internally, it uses our `Portal` primitive which is by default a `div`, however all these components that implement it hardcode `asChild` yet we only pass down portal specific props so there's a discrepancy here.

There are a couple solutions:
- either we forward all props anyway as the merging should be handled by `Slot`
- or we change the types to be only the portal specific props

@andy-hook curious where you're leaning on this one?

**Reply by andy-hook** [reply]
I don't have an issue with forwarding all the props to be honest. Are there any gotchas in doing that vs enforcing only the portal props?

**Reply by benoitgrelard** [reply]
Actually I think this is why we only explicitly passed the Portal props, because otherwise which element should these other props land on as portal wraps both overlay and content? The way it works is they are both portaled separately effectively and each of their node becomes the portal node too.

So really I think we can only go for option 2 now that I've looked into it a bit closer.

**Reply by andy-hook** [reply]
Right, good point. Guess that's our answer then https://github.com/radix-ui/primitives/issues/1974
